Compile chars as JS strings again

alfonsogarciacaro opened the pull request:
Also fixes #1524. Some discussion [here](https://github.com/fable-compiler/Fable/issues/1524#issuecomment-414865851).

ncave commented on src/js/fable-core/Char.ts:
```diff
@@ -116,90 +117,84 @@ const isSymbolMask = 0
-  const test = 1 << getUnicodeCategory(cp);
-  return ((test & isWhiteSpaceMask) !== 0)
-    || (0x09 <= cp && cp <= 0x0D) || cp === 0x85 || cp === 0xA0;
+export function isWhiteSpace(s: string, index?: number) {
```
@alfonsogarciacaro Shouldn't `isWhiteSpace` be using `getUnicodeCategory(s, index)` instead of regex?

alfonsogarciacaro commented:
@ncave I was reverting the PR but you're right, I've removed the regex. I just changed a bit the signature of `getUnicodeSignature` so it's not necessary to call `charCodeAt` twice.
